### ISSUE TITLE
Consider `drdid` speedups for `csdid`

### ISSUE THREAD

**Issue by korenmiklos** [friosavila/csdid_drdid#12]
The propensity score regression can be run only once per `g`, does not need to run for each (`g`, `t`). This can drastically speed up the `drdid` calls when there are many treatment periods. For example, if there are treatments in each of 20 years, this would run 20 logits instead of 400.

Can we pass a pre-calculated propensity score to `drdid` in the nested for loops?

**Comment by korenmiklos** [comment]
Indeed. 

It could still be useful to precompute even GxT times if there are many different outcome variables with the same treatment design. But then we would need a way to save these weights across `csdid` calls.

This does not sound like a priority, but can be important for user adoption later. I have an application where the current code would run for about 2.5hrs *per outcome variable*.
